WordPress/Classes/ViewRelated/Aztec/Stock Photos/StockPhotosMedia.swift

```diff
@@ -1,10 +1,72 @@
 import WPMediaPicker
 
+/*** JSON Structure of a StockPhoto object coming from the API ***
+{
+    "ID": "PEXELS-710916",
+    "URL": "https://images.pexels.com/photos/710916/pexels-photo-710916.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940",
+    "title": "pexels-photo-710916.jpeg",
+    "date": "2018-03-28 00:00:00",
+    "name": "pexels-photo-710916.jpeg",
+    "file": "pexels-photo-710916.jpeg",
+    "guid": "{\"url\":\"https:\\/\\/images.pexels.com\\/photos\\/710916\\/pexels-photo-710916.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940\",\"name\":\"pexels-photo-710916.jpeg\",\"title\":\"pexels-photo-710916.jpeg\"}",
+    "height": 1253,
+    "width": 1880,
+    "thumbnails": {
+        "large": "https://images.pexels.com/photos/710916/pexels-photo-710916.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940",
+        "medium": "https://images.pexels.com/photos/710916/pexels-photo-710916.jpeg?auto=compress&cs=tinysrgb&h=350",
+        "post-thumbnail": "https://images.pexels.com/photos/710916/pexels-photo-710916.jpeg?auto=compress&cs=tinysrgb&h=130",
+        "thumbnail": "https://images.pexels.com/photos/710916/pexels-photo-710916.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=200&w=280"
+    },
+    "type": "image",
+    "extension": "jpeg"
+}
+*/
+
+struct ThumbnailCollection {
+    let largeURL: URL
+    let mediumURL: URL
+    let postThumbnailURL: URL
+    let thumbnailURL: URL
+}
 
 /// Models a Stock Photo
-final class StockPhotosMedia: NSObject, WPMediaAsset {
+///
+final class StockPhotosMedia: NSObject {
+    let id: Int
+    let guid: String
+    let URL: URL
+    let title: String
+    let name: String
+    let size: CGSize
+    let thumbnails: ThumbnailCollection
+
+    init(id: Int, guid: String, URL: URL, title: String, name: String, size: CGSize, thumbnails: ThumbnailCollection) {
+        self.id = id
+        self.guid = guid
+        self.URL = URL
+        self.title = title
+        self.name = name
+        self.size = size
+        self.thumbnails = thumbnails
+    }
+}
+
+extension StockPhotosMedia: WPMediaAsset {
     func image(with size: CGSize, completionHandler: @escaping WPMediaImageBlock) -> WPMediaRequestID {
-        return 0
+
+        //Temporary loading image from URL
+        DispatchQueue.global().async {
```

Comment:
This is just a question, but I would assume there would be already some facility in the codebase to download images, do you know if that's the case?. In any case, even if there was something already, I would not consider it a blocker, we don't really know, at this stage, if this is going to be the final state of this code

Reply:
That is temporary to show the image in the demo. I was thinking on doing something more interesting after having the API working. I checked out the Media Library images and the way they are downloaded is specific for those items.

Maybe we can add some catching too, there will be a lot of reloading. At least a short catching that could last one "search session".

Reply:
@ctarda and @etoledom check `WPImageSource` it's not great but it's already there on the codebase.

Saying that for a long time there has been discussion to replace it with something like [AlamoFire Image cache subspec](https://github.com/Alamofire/AlamofireImage). If you want to have a stab on that it will be great.